components/Util/NumericInput.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState, Dispatch, SetStateAction } from 'react'
+
+export default function NumericInput({
+  label,
+  name,
+  max,
+  min,
+  step,
+  defaultValue,
+  value,
+  onChange,
+}: {
+  label: string
+  name?: string
+  max?: number
+  min?: number
+  step?: number
+  defaultValue?: number
+  value?: number
+  onChange?: Dispatch<SetStateAction<number>>
+}) {
+  // The logical operation here simulates XOR operation
+  // Only either of them should be TRUE at one time.
+  // This operation ensures that
+  if ((value && onChange) === defaultValue) {
+    throw new Error('Provide only either of value or defaultValue.')
+  }
+
+  const isControlled = value !== undefined && onChange !== undefined
+
+  const [internalValue, setInternelValue] = useState(
+    isControlled ? value : defaultValue
+  )
+
+  function handleChange(evt) {
+    const newValue = evt.target.valueAsNumber
+
+    if (isControlled) {
+      onChange(newValue)
+    } else {
+      setInternelValue(newValue)
+    }
+  }
+
+  useEffect(() => {
+    if (isControlled) {
+      setInternelValue(value)
+    }
+  }, [isControlled, value])
+
+  return (
+    <input
```

Comment:
I think we should add onkeydown for the enter key, the same way we have in ui/lesson/Terminal.tsx.

Reply:
Seems reasonable to me.

Since we don't have any place to accept the numericInput, let me create a dummy handler that logs the value in the terminal when "Enter" is pressed.

Reply:
Also @vinayaksh42 

> I am unable to see the Uncontrolled NumericInput component in Vercel deployment link

I opened the Vercel preview [link](https://saving-satoshi-h2m4c54u1-savingsatoshi.vercel.app/) (Mac, Chrome) in both new and incognito windows and can see both components.

<img width="1507" alt="image" src="https://github.com/saving-satoshi/saving-satoshi/assets/85434418/dedf10b9-01bd-461b-b905-c8d491be80a0">

---

Can it be browser specific problem? Which browser are you working with?